### PR TITLE
fcitx5-fluent: init at 0.4.0-unstable-2024-03-30

### PR DESCRIPTION
**Pull request by StarryReverie**
## Description of changes

`fcitx5-fluent` is a fluent-design fcitx5 theme with blur effect and shadow. Homepage: https://github.com/Reverier-Xu/Fluent-fcitx5.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/by-name/fc/fcitx5-fluent/package.nix [hunk 1]
```diff
@@ -0,0 +1,33 @@
+{ lib
+, stdenvNoCC
+, fetchFromGitHub
+}:
+
+stdenvNoCC.mkDerivation rec {
+  pname = "fcitx5-fluent";
+  version = "2024-03-30";
```
Comment (Vonfry) [inline pkgs/by-name/fc/fcitx5-fluent/package.nix:8]:
```suggestion
  version = "unstable-2024-03-30";
```

**Review by Vonfry — APPROVED** [review]

**Review by zendo — APPROVED** [review]

File: pkgs/by-name/fc/fcitx5-fluent/package.nix [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+        cp -r $variant/{theme.conf,panel.png} $out/share/fcitx5/themes/$variant
+
+        for asset in arrow.png next.png back.png radio.png ; do
+            if [[ $(identical $variant/$asset $out/common/$asset) -eq 0 ]]; then
+                ln -s $out/common/$asset $out/share/fcitx5/themes/$variant/$asset
+            else
+                cp -r $variant/$asset $out/share/fcitx5/themes/$variant/$asset
+            fi
+        done
+    done
```
Comment (SuperSandro2000) [inline pkgs/by-name/fc/fcitx5-fluent/package.nix:46]: We use 2 spaces to intend in nix files even for shell code

File: pkgs/by-name/fc/fcitx5-fluent/package.nix [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+
+    runHook postInstall
+  '';
+
+  meta = with lib; {
+    description = "A fluent-design theme with blur effect and shadow";
```
Comment (SuperSandro2000) [inline pkgs/by-name/fc/fcitx5-fluent/package.nix:52]:
```suggestion
    description = "Fluent-design theme with blur effect and shadow";
```

File: pkgs/by-name/fc/fcitx5-fluent/package.nix [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+  version = "0.4.0-unstable-2024-03-30";
+
+  src = fetchFromGitHub {
+    owner = "Reverier-Xu";
+    repo = "Fluent-fcitx5";
+    rev = "dc98bc13e8eadabed7530a68706f0a2a0a07340e";
+    hash = "sha256-d1Y0MUOofBxwyeoXxUzQHrngL1qnL3TMa5DhDki7Pk8=";
+  };
+
+  installPhase = ''
```
Comment (SuperSandro2000) [inline pkgs/by-name/fc/fcitx5-fluent/package.nix:17]: Please consider using something like jdupes instead https://github.com/NixOS/nixpkgs/blob/21576aba43e604a95cc45c25248f39f3989ebe36/pkgs/by-name/fl/fluent-gtk-theme/package.nix#L96

**Pull request merged**